azure/test/client.test.js

```diff
@@ -257,6 +257,48 @@ describe('Azure Log Monitoring', function() {
             testHandleLogs(record, expected, true);
         });
 
+        it('should handle buffer array properly', function() {
+            record = [Buffer.from('{"records": [{ "test": "testing"}]}')];
+            expected = [{ test: 'testing' }];
+            assert.equal(
+                EventhubLogForwarderInstance.getLogFormat(record),
+                constants.BUFFER_ARRAY
+            );
+            testHandleLogs(record, expected, true);
+        });
+
+        it('should handle buffer array without records properly', function() {
+            record = [Buffer.from('{[{ "test": "testing"}]}')];
```

Comment:
This isn't actually valid json input, so it's sending this one as a string log instead of json. I think what's happening is that the spies are created once, so with the test above this testing with the same expected output, the spy is saying it was called with this input but that was because it was called above (hard to explain over text, can explain in person tomorrow)

to test this, you can change "testing" to "ricky" and it'll fail. i'll put up a small pr to fix this.